### ISSUE TITLE
Show synced repositories in add/edit external services UI

### ISSUE THREAD

**Issue by beyang** [sourcegraph/sourcegraph#2866]
Show synced repositories in when adding/editing an external services in this area:

![Selection_056](https://user-images.githubusercontent.com/1646931/54711340-f099a980-4b06-11e9-9879-0280e4ab2763.png)

Feature notes:
* Live-update list of repos ideal when user changes config (but need to check whether this is possible). Otherwise, list should update after user clicks "Update external service" and "Add external service" UI should perhaps include another button that triggers an update.

Related / blocked on:
* https://github.com/sourcegraph/sourcegraph/issues/2574
* https://github.com/sourcegraph/sourcegraph/issues/2025

TODO:
- [ ] Chat with @keegancsmith and @tsenart about how to query server for repositories associated with the given external service.
- [ ] Chat with @francisschmaltz about design

cc @sqs

**Comment by tsenart** [comment]
I believe @keegancsmith and I should own the backend aspect of this feature as it is at the very core of the functionality we're implementing in the #2025. 

We've been describing this as a "dry-run" sync, where we'd compute a Diff of repos like a full sync would, but don't actually make any changes to the database.

This would be a very simple feature to implement on our side once all code hosts are migrated to the new syncer, which is planned for 3.3.

The endpoint to do a full sync of a given external service kind [already exists](https://github.com/sourcegraph/sourcegraph/blob/master/cmd/repo-updater/repoupdater/server.go#L40). It needs to be extended with the dry run mode which is relatively simple.

**Comment by beyang** [comment]
Cool, @keegancsmith do you have a rough ETA of when the dry-run mode will be merged?

**Comment by tsenart** [comment]
@beyang: We can make it a stretch goal for our 3.3 plans. Since we're implementing one code host at a time, you'd have this functionality first for GitHub ready sometime next week. Other codehosts would follow.

**Comment by tsenart** [comment]
@beyang: Do you plan to do this in 3.4? If so, we need to account for it in our own planning for 3.4 to implement dry-run mode.

**Comment by francisschmaltz** [comment]
@beyang can you schedule a meeting so we can discuss the UI

**Comment by beyang** [comment]
@tsenart yes, I was planning to work on it in 3.4, pending the availability of dry-run mode.

**Comment by tsenart** [comment]
@beyang: I'll talk with @keegancsmith when he's back about this and how it could or not fit in 3.4. We should have brought this up during planning, sorry.

**Comment by beyang** [comment]
np, happy to roll w/ w/e you guys decide

**Comment by keegancsmith** [comment]
I am guessing this is backlogged until @beyang has bandwidth to code more :)

**Comment by beyang** [comment]
cc @christinaforney @sqs if this is a product priority, we could talk about someone else to take this on
